git-checkout-plugin fails on windows

Hi there,

your plugin seems pretty nice, sadly it is not working on Windows, since "rm" is not existing.
Is it possible to make the plugin operation system independent? (Maybe with java methods:
https://stackoverflow.com/questions/22918630/java-delete-file-directory-equivalent-of-linux-rm-rf-tmp-garbagedir)

[ERROR] Failed to execute goal com.github.gastaldi:git-sparse-checkout-plugin:1.0.0.Alpha1:git-checkout (default-cli) on project pkp-web: Caught IOException in mojo: Cannot run program "rm" (in directory "D:\dev\pkp\pkp-web\target"): CreateProcess error=2, Das System kann die angegebene Datei nicht finden -> [Help 1]

You were faster than me :-D,
but sadly it didn't work:

[ERROR] Failed to execute goal com.github.gastaldi:git-sparse-checkout-plugin:1.0.0.Alpha2:git-checkout (default-cli) on project pkp-web: Caught IOException in mojo: ...\.git\objects\pack\pack-8a1699ea692bff9c6d0b678a43f2e60834404792.idx -> [Help 1]

Maybe use FileUtils.forceDelete(outputDirectory.toPath().resolve(".git")); instead? (That worked for me)

--- a/src/main/java/com/github/gastaldi/git/GitCheckoutMojo.java
+++ b/src/main/java/com/github/gastaldi/git/GitCheckoutMojo.java
@@ -2,12 +2,8 @@ package com.github.gastaldi.git;
 
 import java.io.File;
 import java.io.IOException;
-import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
-import java.nio.file.SimpleFileVisitor;
-import java.nio.file.attribute.BasicFileAttributes;
 import java.util.List;
 
 import org.apache.maven.plugin.AbstractMojo;
@@ -15,6 +11,7 @@ import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.util.FileUtils;
 
 /**
  * Clones only specific paths from a Git repository
@@ -40,7 +37,8 @@ public class GitCheckoutMojo extends AbstractMojo {
     public void execute() throws MojoExecutionException, MojoFailureException {
         try {
             outputDirectory.mkdirs();
-            if (Files.exists(outputDirectory.toPath().resolve(".git"))) {
+            Path pathToGitDirectory = outputDirectory.toPath().resolve(".git");
+            if (Files.exists(pathToGitDirectory)) {
                 throw new MojoExecutionException("Cannot execute mojo in a directory that already contains a .git directory");
             }
             executeCommand(outputDirectory, "git", "init");
@@ -49,7 +47,7 @@ public class GitCheckoutMojo extends AbstractMojo {
             Path sparseCheckoutFile = outputDirectory.toPath().resolve(".git/info/sparse-checkout");
             Files.write(sparseCheckoutFile, paths);
             executeCommand(outputDirectory, "git", "pull", "origin", branch);
-            deleteDirectory(outputDirectory.toPath().resolve(".git"));
+            FileUtils.forceDelete(pathToGitDirectory.toFile());
             getLog().info("Files were checked out in: " + outputDirectory);
         } catch (IOException e) {
             throw new MojoFailureException("Caught IOException in mojo", e);
@@ -66,34 +64,6 @@ public class GitCheckoutMojo extends AbstractMojo {
             return process.waitFor();
         } catch (InterruptedException e) {
             throw new IOException(e);
-        }
-    }
-
-    /**
-     * Deletes a directory recursively
-     *
-     * @param directory
-     * @throws IOException
-     */
-    private static void deleteDirectory(Path directory) throws IOException {
-        if (Files.exists(directory)) {
-            Files.walkFileTree(directory, new SimpleFileVisitor<Path>() {
-                @Override
-                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                    try {
-                        Files.delete(file);
-                    } catch (NoSuchFileException ignore) {
-                        // Ignore if file is already removed by other process
-                    }
-                    return FileVisitResult.CONTINUE;
-                }
-
-                @Override
-                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
-                    Files.delete(dir);
-                    return FileVisitResult.CONTINUE;
-                }
-            });
         }
     }
 }

